ext/AvizExt/viz/sensitivity.jl

```diff
@@ -235,14 +257,253 @@ function ADRIA.viz.rsa!(g::Union{GridLayout,GridPosition}, rs::ResultSet, si::Na
+    axis_opts::Dict=Dict(),
+)
     f = Figure(; fig_opts...)
     g = f[1, 1] = GridLayout()
     ADRIA.viz.rsa!(g, rs, si, factors; opts, axis_opts)
 
     return f
 end
 
+"""
+    _series_convergence(g::GridPosition, Si_conv::NamedDimsArray, factors::Vector{Symbol};
+        opts::Dict=Dict(:plot_overlay => true), axis_opts::Dict=Dict())
+
+Plot sensitivity values for an increasing number of scenarios as a series, with each member
+    of the series representing a factor or model component.
+
+# Arguments
+- `Si_conv` : Produced using ADRIA.analysis.convergence()
+- `factors` : Factors/model components to plot
+- `opts` : Additional figure customization options
+        - `plot_overlay` : true, to plot overlayed series, false to plot series as grid of subplots
+- `axis_opts` : Additional options to pass to adjust Axis attributes
+      See: https://docs.makie.org/v0.19/api/index.html#Axis
+
+# Returns
+GLMakie figure
+"""
+function _series_convergence(
+    g::GridPosition,
+    Si_conv::NamedDimsArray,
+    factors::Vector{Symbol};
+    opts::Dict=Dict(:plot_overlay => true),
+    axis_opts::Dict=Dict(),
+)
+    plot_overlay = get(opts, :plot_overlay, true)
+    n_scenarios = Si_conv.n_scenarios
+    grps = Dict(Symbol(foi_grp) => foi_grp .== factors for foi_grp in factors)
+
+    xlabel = pop!(axis_opts, :xlabel, "N scenarios")
+    ylabel = pop!(axis_opts, :ylabel, "Pawn Index")
+
+    if :title in keys(axis_opts)
+        title_val = pop!(axis_opts, :title)
+    end
+
+    if plot_overlay
+        ax = Axis(g; axis_opts...)
+        scenarios_confint!(
+            ax,
+            permutedims(Si_conv(; Si=[:lb, :median, :ub]), (3, 1, 2)),
+            grps;
+            x_vals=n_scenarios,
+            sort_by=:none,
+        )
+        ax.xlabel = xlabel
+        ax.ylabel = ylabel
+
+        if @isdefined(title_val)
+            ax.title = title_val
+        end
+    else
+        _colors::Dict{Symbol,Union{Symbol,RGBA{Float32}}} = colors(grps)
+        _alphas::Dict{Symbol,Float64} = alphas(grps)
+
+        n_factors::Int64 = length(factors)
+        if n_factors > 30
+            ArgumentError("Too many factors to plot. Maximum number supported is 30.")
+        end
+
+        n_rows, n_cols = _calc_gridsize(n_factors)
+        factor_names = String.(factors)
+        axs = Axis[]
+        step::Int64 = 1
+
+        for row in 1:n_rows, col in 1:n_cols
+            ax::Axis = Axis(g[row, col]; title=factor_names[step], axis_opts...)
+
+            lines!(
+                ax,
+                n_scenarios,
+                Si_conv(; Si=:median)(; factors=factors[step]);
+                color=(_colors[factors[step]], _alphas[factors[step]]),
+            )
+            band!(
+                ax,
+                n_scenarios,
+                Si_conv(; Si=:lb)(; factors=factors[step]),
+                Si_conv(; Si=:ub)(; factors=factors[step]);
+                color=(_colors[factors[step]], _alphas[factors[step]]),
+            )
```

Comment:
I say drop the bands for now.

This is showing the upper/lower bounds of the results at conditioning points. There are only 10 of these, so the bounds shown here is likely misleading.

The usual approach is to bootstrap the entire analysis.

Reply:
So just plot the lines?

Reply:
Ehm, on second thought maybe put in a big warning about what the bounds are showing?

We'll want to add bootstrapping at some point.

The figures are showing something weird at the moment though

Reply:
![image](https://github.com/open-AIMS/ADRIA.jl/assets/4075136/e48515ca-d33b-4a3b-a4d5-561a1ee99161)


It seems the upper/lower bounds are swapping places?


![image](https://github.com/open-AIMS/ADRIA.jl/assets/4075136/90f3fecc-78ad-4e88-9f75-6624c95945e0)

Reply:
ooh that's weird, let me check it

Reply:
From what I can see the lowera nd upper bounds are giving some weird values, for example:
```julia
infil> Si_conv(; Si=[:lb, :median, :ub])
3-dimensional NamedDimsArray(KeyedArray(...)) with keys:
↓   factors ∈ 3-element Vector{Symbol}
→   Si ∈ 3-element view(::Vector{Symbol},...)
◪   n_scenarios ∈ 10-element Vector{Int64}
And data, 3×3×10 view(::Array{Float64, 3}, :, [2, 4, 5], :) with eltype Float64:
[showing 3 of 10 slices]
[:, :, 1] ~ (:, :, 25):
                    (:lb)      (:median)  (:ub)
  (:dhw_scenario)    0.765149   0.871221   1.0
  (:wave_scenario)   0.144206   0.631884   0.58991
  (:guided)          0.172928   0.705589   0.755389

[:, :, 6] ~ (:, :, 150):
                    (:lb)      (:median)  (:ub)
  (:dhw_scenario)    0.558062   0.629598   0.837923
  (:wave_scenario)   0.603173   0.728472   0.45569
  (:guided)          0.657582   0.591617   0.714464

[:, :, 10] ~ (:, :, 250):
                    (:lb)      (:median)  (:ub)
  (:dhw_scenario)    0.54634    0.573742   0.743484
  (:wave_scenario)   0.554806   0.56338    0.478
  (:guided)          0.563533   0.605634   0.66728
```
In this case the upper bound is less than the median for `:dhw_scenario`. There are some similar anomalies for other variables. Could this happen if the sample size is too small maybe? I couldn't see anything wrong with how the lower and upper bounds are calculated in `pawn()`. I haven't done a very large sample yet but I can try that to see if it is the issue.

Reply:
I looked into it further- it's the column-wise normalisation which is causing this in `analysis.convergence`.  Because we normalise relative to other factors the normalised upper bound can be lower than the normalised median. Maybe I just do median +/-std*0.5 or something for now and implement bootstrapping later?

Reply:
Okay, I think drop the column normalization - it's inappropriate in this case. The col-norm was purely so you could assess factor ranks for each statistic of interest. I should have caught that earlier, sorry.

Reply:
Thanks, I'll do that